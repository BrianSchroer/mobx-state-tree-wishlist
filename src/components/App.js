import React, { Component } from 'react';
import logo from '../assets/santa-claus.png';
import WishListView from './WishListView';

class App extends Component {
  state = {
    selectedUserId: null
  };

  render() {
    const { group } = this.props;
    const selectedUser = group.users.get(this.state.selectedUserId);

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Wish List</h1>
        </header>

        <select onChange={this.onSelectUser}>
          <option>- Select user -</option>
          {group.users.values().map(user => (
            <option key={user.id} value={user.id}>
              {user.name}
            </option>
          ))}
        </select>

        {selectedUser && <WishListView wishList={selectedUser.wishList} />}
      </div>
    );
  }

  onSelectUser = event => {
    this.setState({ selectedUserId: event.target.value });
  };
}

export default App;
