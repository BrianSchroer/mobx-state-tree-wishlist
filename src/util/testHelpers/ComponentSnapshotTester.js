import React from 'react';
import renderer from 'react-test-renderer';

/**
 * React component snapshot helper
 */
export default class ComponentSnapshotTester {

  /**
   * Create new ComponentSnapshotTester instance
   * @param {*} component - React component
   */
  constructor(component) {
    this.component = component.type;
    this.props = component.props;
    this.adjustProps = (props, propOverrides) => Object.assign({}, props, propOverrides);
  }

  /**
   * Specify callback function to be called by the test function to adjust component
   * props with propOverrides.
   * (Defaults to adjusting top-level props.)
   */
  withPropAdjustor = (adjustor) => {
    this.adjustProps = adjustor;
    return this;
  };

  /**
   * Adjust props that were passed to the constructor with propOverrides, then create
   * snapshot and assert that it matches previously accepted screenshot.
   */
  test = (propOverrides) => {
    const Component = this.component;
    const props = this.adjustProps(this.props, propOverrides);

    const newSnapshot = renderer.create(<Component {...props} />).toJSON();

    expect(newSnapshot).toMatchSnapshot();
  }

  /**
   * Create snapshot and assert that it matches previously accepted screenshot. 
   * @param {*} component - React component1 
   */
  static test(component) {
    new ComponentSnapshotTester(component).test();
  }
}