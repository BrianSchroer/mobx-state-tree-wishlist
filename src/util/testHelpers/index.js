export { default as mobxSnapshotHelper } from './mobxSnapshotHelper/snapshotHelper';
export { default as ComponentSnapshotHelper } from './ComponentSnapshotHelper/SnapshotHelper';