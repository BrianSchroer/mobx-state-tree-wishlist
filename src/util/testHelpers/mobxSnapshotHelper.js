import { getSnapshot } from 'mobx-state-tree';

export default class {
  static createSnapshot(target) {
    return getSnapshot(target);
  }

  static assertMatch(target) {
    expect(this.createSnapshot(target)).toMatchSnapshot();
  }
}