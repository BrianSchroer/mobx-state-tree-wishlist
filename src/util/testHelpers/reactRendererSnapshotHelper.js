import renderer from 'react-test-renderer';

/**
 * react-test-renderer snapshot helper methods
 */
export default class {

  /**
   * Create component from JSX
   * @param {JSX} componentJsx (e.g. '<Button />')
   * @returns {object} component object
   */
  static createComponent(componentJsx) {
    return renderer.create(componentJsx);
  }

  /**
   * Create component from JSX and serialize as JSON "snapshot" string.
   * @param {JSX} componentJsx (e.g. '<Button />')
   * @returns {string} JSON string.
   */
  static createSnapshot(componentJsx) {
    return this.createComponent(componentJsx).toJSON();
  }

  /**
   * Create component from JSX, serialize as JSON string, and compare to accepted snapshot.
   * If a snapshot doesn't already exist, one is saved. If a previous snapshot exists
   * and the new snapshot doesn't match, you'll be asked if that's an error or if you want
   * to accept the new snapshot.
   * @param {JSX} componentJsx (e.g. '<Button />')
   */
  static assertMatch(componentJsx) {
    expect(this.createSnapshot(componentJsx)).toMatchSnapshot();
  }
}