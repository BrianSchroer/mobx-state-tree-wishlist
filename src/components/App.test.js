import React from 'react';
import App from './App';
import { reactRendererSnapshotHelper as snapshotHelper } from '../util/testHelpers';

const items = [1, 2, 3].map(i => ({
  name: `name${i}`,
  price: i,
  image: `http://www.something/com/image${i}.png`,
  remove: () => { }
}));

const wishList = {
  items,
  totalPrice: 1234.56
};

describe('App', () => {
  it('should render correctly', () => {
    snapshotHelper.assertMatch(<App wishList={wishList} />);
  });
});