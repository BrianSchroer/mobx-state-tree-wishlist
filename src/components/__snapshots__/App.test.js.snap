// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="santa-claus.png"
    />
    <h1
      className="App-title"
    >
      Wish List
    </h1>
  </header>
  <div
    className="list"
  >
    <ul>
      <li
        className="item"
      >
        <img
          alt="name1"
          src="http://www.something/com/image1.png"
        />
        <h3>
          name1
        </h3>
        <span>
          $1.00
        </span>
        <span>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Edit"
              role="img"
            >
              ✏️
            </span>
          </button>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Remove"
              role="img"
            >
              ❌
            </span>
          </button>
        </span>
      </li>
      <li
        className="item"
      >
        <img
          alt="name2"
          src="http://www.something/com/image2.png"
        />
        <h3>
          name2
        </h3>
        <span>
          $2.00
        </span>
        <span>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Edit"
              role="img"
            >
              ✏️
            </span>
          </button>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Remove"
              role="img"
            >
              ❌
            </span>
          </button>
        </span>
      </li>
      <li
        className="item"
      >
        <img
          alt="name3"
          src="http://www.something/com/image3.png"
        />
        <h3>
          name3
        </h3>
        <span>
          $3.00
        </span>
        <span>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Edit"
              role="img"
            >
              ✏️
            </span>
          </button>
          <button
            onClick={[Function]}
          >
            <span
              aria-label="Remove"
              role="img"
            >
              ❌
            </span>
          </button>
        </span>
      </li>
    </ul>
    <div>
      Total: 
      $1,234.56
    </div>
    <div
      style={
        Object {
          "marginTop": "2em",
        }
      }
    >
      <div>
        <div
          className="item-edit"
        >
          <div>
             Thing: 
            <input
              onChange={[Function]}
              value=""
            />
             
          </div>
          <div>
             Price: 
            <input
              onChange={[Function]}
              value={0}
            />
             
          </div>
          <div>
             Image: 
            <input
              onChange={[Function]}
              value=""
            />
             
          </div>
        </div>
        <button
          onClick={[Function]}
        >
          <span
            aria-label="Add"
            role="img"
          >
            Add
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
